src/operator/src/expr_factory.rs

```diff
@@ -281,6 +281,18 @@ pub fn validate_create_expr(create: &CreateTableExpr) -> Result<()> {
         }
         .fail();
     }
+    // verify do not contain interval type column issue #3235
+    for column in &create.column_defs {
+        if matches!(column.data_type(), ColumnDataType::IntervalMonthDayNano) {
```

Comment:
We may match all interval types.
```suggestion
        if matches!(column.data_type(), ColumnDataType::IntervalYearMonth | ColumnDataType::IntervalDayTime | ColumnDataType::IntervalMonthDayNano) {
```